lib/jsonapi/view.ex

```diff
@@ -94,23 +96,38 @@ defmodule JSONAPI.View do
         do: serialize(__MODULE__, models, conn)
 
       def url_for(nil, nil) do
-        "#{@namespace}/#{type()}"
+        case @host do
```

Comment:
Are all of these case statements really necessary?  This is already inside of a macro, why not only generate the methods that are necessary and avoid the necessary `case`?